graph/src/task_spawn.rs

```diff
@@ -48,20 +43,14 @@ pub fn spawn_blocking<T: Send + 'static>(
     tokio::task::spawn_blocking(move || block_on(abort_on_panic(f)))
 }
 
-/// Panics result in an `Err` in `JoinHandle`.
-pub fn spawn_blocking_allow_panic<T: Send + 'static>(
-    f: impl Future03<Output = T> + Send + 'static,
-) -> JoinHandle<T> {
-    tokio::task::spawn_blocking(move || block_on(f))
-}
-
-/// Does not abort on panic
-pub async fn spawn_blocking_async_allow_panic<R: 'static + Send>(
+/// Does not abort on panic, panics result in an `Err` in `JoinHandle`.
+pub fn spawn_blocking_allow_panic<R: 'static + Send>(
     f: impl 'static + FnOnce() -> R + Send,
-) -> R {
-    tokio::task::spawn_blocking(f).await.unwrap()
+) -> JoinHandle<R> {
+    tokio::task::spawn_blocking(f)
```

Comment:
Futures generally shouldn't be blocking, `spawn_blocking_allow_panic` had few uses and they could be refactored to use ordinary closures instead of futures.